src/Result.php

```diff
@@ -50,11 +50,25 @@ public function jsonSerialize(): mixed
         }
 
         if ($this->userOutput !== '') {
-            $result['output'] = $this->userOutput;
+            // In 2024 some innocent ASCII code still fools displays and editors.
+            $result['output'] = \str_replace(
+                [
+                    "\u{7F}", // Delete
+                ],
+                "\u{FFFD}", // Unicode substitute for invalid characters
+                $this->userOutput
+            );
```

Comment:
Do we need this if we have `JSON_INVALID_UTF8_SUBSTITUTE`?

Reply:
Yes, because this is valid UTF-8. It is not handled by PHPUnit (which turns other occurences of invisible characters < 0x1F into hexadecimal string representation (`bin2hex()`). But this characters eats the character after it, and so becomes a debugging nightmare for beginners. And me.

Reply:
Thanks for the explanation! Would it be better to cast it to a unicode representation instead of an invalid character?

Reply:
I'm not sure if `U+2421 ␡` is actually more helpful than `U+FFFD �` to students. In my terminal the Unicode graphical representation ␡ does not always show up. It is displayed by `cat` when used in the PHP file, but not printed when running the tests in Docker. It ends up in the JSON correctly encoded. But there is quirks, and I'd stick with the very old � for our purpose ("this char is not expected to be here").